File: ios/Classes/Bridges/LogLevelBridge.swift
import Flutter
import SuperwallKit

// TODO
public class LogLevelBridge: BridgeInstance {
  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    guard let args = call.arguments as? [String: Any],
          let levelValue = args["level"] as? Int,
          let logLevel = LogLevel(rawValue: levelValue) else {
      result(FlutterError(code: "INVALID_ARGUMENTS", message: "Invalid log level", details: nil))
      return
    }

    switch call.method {
      case "getLogLevelDescription":
        result(logLevel.description)
      case "getLogLevelDescriptionEmoji":
        result(logLevel.descriptionEmoji)
      default:
        result(FlutterMethodNotImplemented)
    }
  }
}

File: android/src/main/kotlin/com/superwall/superwallkit_flutter/bridges/LogLevelBridge.kt
package com.superwall.superwallkit_flutter.bridges

import LogLevel
import android.content.Context
import com.superwall.superwallkit_flutter.badArgs
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.Result

class LogLevelBridge(
    context: Context,
    bridgeId: BridgeId,
    initializationArgs: Map<String, Any>? = null
) : BridgeInstance(context, bridgeId, initializationArgs) {
    override fun onMethodCall(call: MethodCall, result: Result) {
        val levelValue = call.argument<Int>("level")
        val logLevel = levelValue?.let { findLogLevel(it) }

        if (logLevel != null) {
            when (call.method) {
                "getLogLevelDescription" -> result.success(logLevel.toString())
                "getLogLevelDescriptionEmoji" -> result.success(logLevel.getDescriptionEmoji())
                else -> result.notImplemented()
            }
        } else {
            result.badArgs(call);
        }
    }

    private fun findLogLevel(value: Int): LogLevel? {
        return LogLevel.values().firstOrNull { it.level == value }
    }
}
